autorest/codegen/models/parameter_list.py

```diff
@@ -105,6 +105,7 @@ def content_type_parameter(self) -> Parameter:
                     if p.rest_api_name == "Content-Type"
                 ]
             ))
+            content_type_param.is_kwarg = True
```

Comment:
the reason why I've moved this setting to here is to prevent a breaking change. Basically, if your operation doesn't have a body, we allowed you to define a content type param and generate it as positional, see below:
<img width="1653" alt="Screen Shot 2021-06-22 at 1 31 31 PM" src="https://user-images.githubusercontent.com/43154838/122973569-608ae000-d35f-11eb-92b2-617158047a62.png">

Now, this was in a test, and idt anyone actually uses it, but I thought we might as well not break this situation, and only set content type param to a kwarg in the case where the operation actually has a body. This behavior is not breaking in this case, bc we weren't able to generate before

Reply:
Additionally, in the version tolerant autorest, Content-Type will be a keyword only param bc it's a header, so the version tolerant behavior will eventually have it as keyword only